server/bin/www

```diff
@@ -33,15 +34,27 @@ const io = require("socket.io")(server);
 
 io.on("connection", (socket) => {
   socket.on("go-online", (id) => {
+    console.info(`Client connected [id=${socket.id}]`);
     if (!onlineUsers.includes(id)) {
       onlineUsers.push(id);
+      userMap[id] = socket.id
+      console.log(userMap)
     }
     // send the user who just went online to everyone else who is already online
     socket.broadcast.emit("add-online-user", id);
   });
 
   socket.on("new-message", (data) => {
-    socket.broadcast.emit("new-message", {
+    console.log(data)
+    const senderId = data.sender.id;
+    const recipientId = data.recipientId;
+
+    socket.to(userMap[senderId]).broadcast.emit("new-message", {
+      message: data.message,
+      recipientId: data.recipientId,
+      sender: data.sender,
+    });
```

Comment:
The user sending the message shouldn't need to receive the message; they can just update the local state with the message they're sending.